From a5cddd0e657b9fc3f6775da8ebdaa6d25446b649 Mon Sep 17 00:00:00 2001
From: Alex Fan <alex.fan.q@gmail.com>
Date: Sat, 7 Jan 2023 23:26:10 +1100
Subject: [PATCH] fix implicit-function-declaration error

clang-16 (and future gcc) makes implicit-function-declaration an error
add syscall() declaration by #include <unistd.h>

Signed-off-by: Alex Fan <alex.fan.q@gmail.com>
---
 src/thread_id.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/thread_id.h b/src/thread_id.h
index bb3875c..03ee5d3 100644
--- a/src/thread_id.h
+++ b/src/thread_id.h
@@ -7,6 +7,7 @@
 #if JSC_THREAD_ID_METHOD == 1
 
 #include <sys/syscall.h>
+#include <unistd.h>
 
 #define GET_THREAD_ID syscall(SYS_gettid)
 
